spring-cloud-sleuth-zipkin-stream/src/test/java/org/springframework/cloud/sleuth/zipkin/stream/SamplingZipkinSpanIteratorTests.java

```diff
@@ -76,6 +79,54 @@ public boolean isSampled(long l) {
 				"message:foo", "message:baz");
 	}
 
+	@Test
+	public void appendsLocalComponentTagIfNoZipkinLogIsPresent() {
+		Spans spans = new Spans(this.host, Collections.singletonList(span("foo")));
+
+		Iterator<zipkin.Span> result = new SamplingZipkinSpanIterator(
+				Sampler.create(1.0f), spans);
+
+		assertThat(result)
+				.flatExtracting(s -> s.binaryAnnotations)
+				.extracting(input -> input.key)
+				.contains(Constants.LOCAL_COMPONENT);
+	}
+
+	@Test
+	public void appendsServerAddressTagIfClientLogIsPresent() {
+		Span span = span("foo");
+		span.logEvent(Constants.CLIENT_RECV);
```

Comment:
switch all the tests s/CLIENT_RECV/CLIENT_SEND as it makes more sense (because it would be an error to have only a RECV annotation)